src/SDKs/ServiceBus/ServiceBus.Tests/Tests/ScenarioTests.NamespaceTests.Detail.CRUD.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace ServiceBus.Tests.ScenarioTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Management.ServiceBus;
+    using Microsoft.Azure.Management.ServiceBus.Models;
+    using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+    using TestHelper;
+    using Xunit;
+    public partial class ScenarioTests
+    {
+        [Fact]
+        public void NamespaceCreateGetUpdateDeleteDeatil()
+        {
+            using (MockContext context = MockContext.Start(this.GetType().FullName))
+            {
+                InitializeClients(context);
+
+                var location = this.ResourceManagementClient.GetLocationFromProvider();
+
+                var resourceGroup = "v-ajnavtest";
+                //var resourceGroup = this.ResourceManagementClient.TryGetResourceGroup(location);
+                //if (string.IsNullOrWhiteSpace(resourceGroup))
+                //{
+                //    resourceGroup = TestUtilities.GenerateName(ServiceBusManagementHelper.ResourceGroupPrefix);
+                //    this.ResourceManagementClient.TryRegisterResourceGroup(location, resourceGroup);
+                //}
+
+                // Create Invalid Namespace
+                var invalidNamespaceName = TestUtilities.GenerateName(null);
+
+                //Check Invalid namespace name available
+                var checkInvalidNamespaceavailable = ServiceBusManagementClient.Namespaces.CheckNameAvailabilityMethod(new CheckNameAvailability() { Name = invalidNamespaceName });
+
+                Assert.Equal("InvalidName", checkInvalidNamespaceavailable.Reason.ToString());
+
+                // Create Namespace
+                var namespaceName = TestUtilities.GenerateName(ServiceBusManagementHelper.NamespacePrefix);
+
+                //Check namespace name available
+                var checknamespaceavailable = ServiceBusManagementClient.Namespaces.CheckNameAvailabilityMethod(new CheckNameAvailability() { Name = namespaceName });
+
+                var createNamespaceResponse = this.ServiceBusManagementClient.Namespaces.CreateOrUpdate(resourceGroup, namespaceName,
+                    new SBNamespace()
+                    {
+                        Location = location,
+                        Sku = new SBSku
+                        {
+                            Name = SkuName.Standard,
+                            Tier = SkuTier.Standard
+                        },
+                        Tags = new Dictionary<string, string>()
+                        {
+                            {"tag1", "value1"},
+                            {"tag2", "value2"}
+                        }
+                    });
+
+                Assert.NotNull(createNamespaceResponse);
```

Comment:
please verify all properties, like Tags and Sku

Reply:
Done!